features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
Both this instance and the SingletonPtr instance should be allocated globally. According to this post, gcc by default emits thread-safe static constructors, although I'm not sure if we need to tie into hooks for that.

It also raises the question of thread-safety on other toolchains.

If this _is_ garunteed to be thread-safe per the C++03 standard, it may be a better solution over the existing `SingletonPtr` and we may want consider deprecating the `SingletonPtr` class in favor of this method. But that is a big if at the moment.